x/stakeibc/keeper/update_redemption_rates_test.go

```diff
@@ -271,33 +271,55 @@ func (s *KeeperTestSuite) TestGetRedemptionRate_DepositRecords() {
 }
 
 func (s *KeeperTestSuite) TestGetTokenizedDelegation() {
+	transferInProgress := recordtypes.LSMTokenDeposit_TRANSFER_IN_PROGRESS
+	detokenizationQueue := recordtypes.LSMTokenDeposit_DETOKENIZATION_QUEUE
+	detokenizationInProgress := recordtypes.LSMTokenDeposit_DETOKENIZATION_IN_PROGRESS
+	transferFailed := recordtypes.LSMTokenDeposit_TRANSFER_FAILED
+	detokenizationFailed := recordtypes.LSMTokenDeposit_DETOKENIZATION_FAILED
+
+	validators := []*types.Validator{
+		{Address: "valA", InternalSharesToTokensRate: sdk.OneDec()},
+		{Address: "valB", InternalSharesToTokensRate: sdk.MustNewDecFromStr("0.75")},
+		{Address: "valC", InternalSharesToTokensRate: sdk.MustNewDecFromStr("0.5")},
+	}
+
+	// Total: 1 + 2 + 3 + 4 + 5 + 6 + 7 + 8 + 9 + 10 = 65
 	lsmDeposits := []recordtypes.LSMTokenDeposit{
-		// Total: 1 + 2 + 3 + 4 + 5 + 6 + 7 + 8 + 9 + 10 = 65
-		{ChainId: HostChainId, Status: recordtypes.LSMTokenDeposit_TRANSFER_IN_PROGRESS, Amount: sdk.NewInt(1)},
-		{ChainId: HostChainId, Status: recordtypes.LSMTokenDeposit_TRANSFER_IN_PROGRESS, Amount: sdk.NewInt(2)},
+		// ValidatorA Exchange Rate 1.0
+		{ChainId: HostChainId, Status: transferInProgress, Amount: sdk.NewInt(1), ValidatorAddress: "valA"}, // 1 * 1.0 = 1
```

Comment:
Should these tests cover the TRANSFER_QUEUE state?

Reply:
good call, done